lib/public_cocina_record.rb

```diff
@@ -3,27 +3,77 @@
 require 'http'
 
 class PublicCocinaRecord
-  attr_reader :druid, :purl_url
+  attr_reader :public_cocina_doc, :druid, :purl_url
 
-  def self.fetch(url)
-    response = HTTP.get(url)
-    response.body if response.status.ok?
+  def self.fetch(druid, purl_url: 'https://purl.stanford.edu')
+    response = HTTP.get("#{purl_url}/#{druid}.json")
+    new(druid, response.body, purl_url:) if response.status.ok?
   end
 
-  def initialize(druid, purl_url: 'https://purl.stanford.edu')
+  def initialize(druid, public_cocina, purl_url: 'https://purl.stanford.edu')
     @druid = druid
     @purl_url = purl_url
+    @public_cocina_doc = JSON.parse(public_cocina)
+  end
+
+  def public_cocina
+    @public_cocina ||= self.class.fetch("#{purl_url}/#{druid}.json")
```

Comment:
What's the relationship between `@public_cocina_doc` and `@public_cocina`?

Reply:
Hmm. this might be a holdover from #1409. I'm not sure why you'd ever have the thing `fetch` itself. Seems like we only need `public_cocina_doc`?

Reply:
Tests passed without this method, so I removed it.